context/app/static/js/pages/CellsAPIDemo/SearchBySubstring.jsx

```diff
@@ -0,0 +1,64 @@
+import React, { useState } from 'react';
+
+import Autocomplete from '@material-ui/lab/Autocomplete';
+import Paper from '@material-ui/core/Paper';
+import TextField from '@material-ui/core/TextField';
+
+// eslint-disable-next-line no-unused-vars
+function SearchBySubstring(props) {
+  const { targetEntity } = props;
+
+  const [substring, setSubstring] = useState('');
+  const [options, setOptions] = useState([]);
+
+  async function handleChange(event) {
+    const { target } = event;
+    setSubstring(target.value);
+
+    if (target.value === '') {
+      setOptions([]);
+      return;
+    }
+
+    const urlParams = new URLSearchParams();
```

Comment:
Is this the start of making queries accessible by URL? Is there a common pattern for managing this?

Reply:
I'm not sure if I understand your point, but one possibility would be, instead of having all these different URLs scattered in different components, we have a central service with methods that encapsulate the HTTP details. I've just filed #1945.... but if that doesn't get to your concern, let me know?